### PR TITLE
Support period after version core

### PR DESCRIPTION
**Pull request by eneuschild**
Re: https://github.com/go-gitea/gitea/issues/13067

Had to change a little more than initially expected both within the regex itself and in unit tests, but this should do for gitea purposes.

### CODE REVIEW

**Comment by eneuschild** [comment]
Wh00t, conflicts! I'm checking.

**Comment by 6543** [comment]
u hmm #3 or #2 ...

File: version_test.go [hunk 1]
```diff
@@ -10,9 +10,14 @@ func TestNewVersion(t *testing.T) {
 		version string
 		err     bool
 	}{
+		{"2.29.0.rc0.261.g7178c9af9c", false},
+		{"", true},
 		{"1.2.3", false},
```
Comment (6543) [inline version_test.go]:
```suggestion
		// No Error
		{"1.2.3", false},
```

File: version_test.go [hunk 2]
```diff
@@ -10,9 +10,14 @@ func TestNewVersion(t *testing.T) {
...
 		{"1.0", false},
 		{"1", false},
+		{"1.2.beta", false},
+		{"1.21.beta", false},
+		{"foo", true},
```
Comment (6543) [inline version_test.go]:
```suggestion
```

File: version_test.go [hunk 3]
```diff
@@ -31,10 +36,6 @@ func TestNewVersion(t *testing.T) {
 		{"2.28.0.618+gf4bc123cb7", false},
 		{"1.13.0+dev-545-gb3b1c081b", false},
 		{"2.28.0.618.gf4bc123cb7", false},
```
Comment (6543) [inline version_test.go:34]:
```suggestion
		{"2.28.0.618.gf4bc123cb7", false},
		{"2.29.0.rc0.261.g7178c9af9c", false},
		{"1.2.beta", false},
		{"1.21.beta", false},

		// Have Error
		{"", true},
```

**Review by 6543 — COMMENTED** [review]
sort tests ...

File: version_test.go [hunk 4]
```diff
@@ -31,10 +32,12 @@ func TestNewVersion(t *testing.T) {
...
+		{"2.29.0.rc0.261.g7178c9af9c", false},
+		{"1.2.beta", false},
+		{"1.21.beta", false},
+
+		// Have Error
 		{"", true},
```
Comment (6543) [inline version_test.go:40]:
```suggestion
		{"", true},
		{"foo", true},
```

**Review by 6543 — APPROVED** [review]
:rocket:

**Pull request merged**